include/aspect/particle/world.h

```diff
@@ -83,6 +83,7 @@ namespace aspect
          * take the properties of the closest particle, but more complicated
          * schemes are possible.
          *
+         * @param [in] generator The particle generator for this world.
          * @param [in] manager The property manager for this world.
          * @param [in] load_balancing The strategy used to balance the
          * computational load for particle advection across processes.
```

Comment:
I am not exactly sure, what is the right answer to this, I feel like I know when I *can* use unique or shared pointers instead of raw pointers, but not when I *should* use them. 

Let me describe what I am doing: I do not want to add parse_parameters and declare_parameters functions to world.cc as well, so I let the Tracers postprocessor declare and parse input options for the world. Tracers.cc:parse_parameters(prm) reads which plugins to create and creates the plugins, which gives parse_parameters a raw pointer to these objects. These pointers however go out of scope at the end of parse_parameters and after I initialize World with it, therefore I want to transfer ownership of the generated plugins to World during the call to initalize. 

Initially I used raw pointers to hand them over to World, and store them in World. In a review you mentioned I should use unique or shared_ptrs instead (dependings on the purpose), so I went for shared_ptr in Tracers and World (I might have missunderstood you there). Thinking about it again, I probably should have sticked to the raw pointers in tracers.cc:parse_parameters, hand them over as raw pointers to World, and store them as unique pointers in World?